yt_dlp/extractor/vrsquare.py

```diff
@@ -0,0 +1,191 @@
+
+        return {
+            'id': video_id,
+            'title': self._html_search_meta(['og:title', 'twitter:title'], webpage),
+            'description': self._html_search_meta('description', webpage),
+            'formats': self._extract_m3u8_formats(traverse_obj(web_api, (
+                'urls', ..., 'url', any)), video_id, 'mp4', fatal=False),
+            'thumbnail': self._html_search_meta('og:image', webpage),
+            **traverse_obj(webpage, {
+                'duration': ({find_element(cls='layout-product-data-time')}, {parse_duration}),
+                'tags': ({find_elements(cls='search-tag')}, ..., {clean_html}),
+            }),
+        }
+
+
+class VrSquarePlaylistBaseIE(VrSquareIE):
+    def _fetch_vids(self, source, keys=()):
+        return [self.url_result(
+            f'{self._BASE_URL}/contents/{x.removeprefix("/contents/")}', VrSquareIE)
+            for x in traverse_obj(source, (
+                *keys, {find_elements(cls='video', html=True)},
+                ..., {extract_attributes}, 'data-url'))
+        ]
+
+
+class VrSquareChannelIE(VrSquarePlaylistBaseIE):
+    IE_NAME = 'vrsquare:channel'
+
+    _VALID_URL = r'https?://livr\.jp/channel/(?P<id>\w+)'
+    _TESTS = [{
+        'url': 'https://livr.jp/channel/H372648599',
+        'info_dict': {
+            'id': 'H372648599',
+            'title': 'AKB48＋チャンネル',
+        },
+        'playlist_mincount': 502,
+    }]
+
+    def _entries(self, playlist_id):
+        for page in itertools.count(1):
+            ajax = self._download_json(
+                f'{self._BASE_URL}/ajax/channel/{playlist_id}',
+                playlist_id, query={'p': page},
+            )
+            yield from self._fetch_vids(ajax, ('contents_render_list', ...))
+            if not ajax['hasNext']:
+                break
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        webpage = self._download_webpage(url, playlist_id)
+
+        return self.playlist_result(
+            self._entries(playlist_id), playlist_id, self._html_search_meta('og:title', webpage))
+
+
+class VrSquareSearchIE(VrSquarePlaylistBaseIE):
+    IE_NAME = 'vrsquare:search'
+
+    _VALID_URL = r'https?://livr\.jp/web-search'
+    _TESTS = [{
+        'url': 'https://livr.jp/web-search?w=%23%E5%B0%8F%E6%A0%97%E6%9C%89%E4%BB%A5',
+        'info_dict': {
+            'id': '#小栗有以',
+            'title': '#小栗有以',
+        },
+        'playlist_mincount': 70,
+    }]
+
+    def _entries(self, query):
+        for page in itertools.count(1):
+            ajax = self._download_json(
+                f'{self._BASE_URL}/ajax/web-search',
+                query, query={'p': page, 'w': query},
+            )
+            yield from self._fetch_vids(ajax, ('contents_render_list', ...))
+            if not ajax['has_next']:
+                break
+
+    def _real_extract(self, url):
+        query = parse_qs(url).get('w', [None])[0]
```

Comment:
Might as well make this fatal
```suggestion
        query = parse_qs(url)['w'][0]
```